src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -299,6 +300,135 @@ public record {|CS0115:{|CS0115:{|CS0115:{|CS8867:C|}|}|}|}(int P) : {|CS8864:B|
             await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
         }
 
+        [Fact]
+        public async Task TestMovePropertySimpleRecordInheritance()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public record B
+    {
+        public int Foo { get; init; }
+    }
+
+    public class [|C|] : {|CS8865:B|}
+    {
+        public int P { get; init; }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record B
+    {
+        public int Foo { get; init; }
+    }
+
+    public record C(int P) : B;
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+            await TestCodeFixAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
```

Comment:
So this seems a bit odd to me that we'd count this as both a refactoring and a fix at this point -- if we have it registered as a fix, do we also need the refactoring? Isn't that going to make duplicate entries in the light bulb menu or something? I don't know what precedent is here though, @CyrusNajmabadi you might?

Reply:
A couple of notes:
1. Most of the time it is a refactoring. The refactoring only activates on the class declaration.
2. In the much more specific case where we have a class inheriting from a record, this can fix the diagnostic that says that classes can't inherit from records.
3. Writing an analyzer that could tell when the class can be made into a record safely feels like overkill as opposed to just offering the refactoring if we can.
4. The code actions from Refactoring and code fix have the same equivalence key. Not sure if it would help them from duplicating.

Reply:
Can you verify that they don't both show at the same time? Having a code fix does help a lot here because it will show up for the lightbulb next to the squiggle, but it would be odd if they both showed up.

Reply:
Yep! I'm not sure if there's a good way to write a unit test for this but:
![image](https://user-images.githubusercontent.com/43689321/186992737-1d9523b0-569d-4356-bc1f-5f93ca293b81.png)

One note is that the selection spans that would activate the code fix versus the refactoring are slightly different and do not overlap (but may touch each other). The diagnostic and therefore the code fix activates on the base syntax, while the refactoring activates on the class declaration syntax, which does not work on the base syntax (I added a test for that [here](https://github.com/dotnet/roslyn/pull/63525/files#diff-0f334fb44de052f37804a9b2cc14299dc97601a25f39030765c71341d66f75e3R4038)).

In this case I selected both the class declaration and the base syntax, so that if both refactoring and fix would show up at all then it would here. I also tried selecting just the class declaration and just the base syntax as well, and each time there is only one option listed.

Reply:
I think just adding a single test to verify the equivalence keys are the same is sufficient, just so if that changes we have a unit test explaining why they currently are that will break